cloud/src/meta-service/meta_service_resource.cpp

```diff
@@ -509,6 +509,80 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
     LOG(INFO) << vault_msg;
 }
 
+static int alter_storage_vault(InstanceInfoPB& instance, std::unique_ptr<Transaction> txn,
```

Comment:
Add an UT to test it.
And refer to `MetaServiceImpl::alter_obj_store_info` and `MetaServiceImpl::alter_instance` to abstract the behavior and make as flexible as possible.

Reply:
Ut and regression cases would be involved in #37606.